daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/BlobLengthParsers.scala

```diff
@@ -24,38 +24,32 @@ import org.apache.daffodil.processors.ElementRuntimeData
 import org.apache.daffodil.processors.LengthInBitsEv
 import org.apache.daffodil.util.MaybeULong
 
-sealed abstract class BlobLengthParser(override val context: ElementRuntimeData)
-  extends PrimParser {
-
-  protected def getLengthInBits(pstate: PState): Long
-
-  override final def parse(start: PState): Unit = {
+trait ByteChunkWriter { self: PrimParser =>
+
+  /**
+   * Read nBits of data from the input source associated with pstate in chunks.
+   * For each chunk, the writeChunk function is called, which takes as
```

Comment:
Let's make the comment clearer that the writeChunk function is passed as a parameter:

    * For each chunk, the writeChunk function passed as a parameter is called, which takes as